lib/elixir/lib/string.ex

```diff
@@ -2252,11 +2252,82 @@ defmodule String do
   end
 
   @doc """
-  Returns a float value between 0 (equates to no similarity) and 1 (is an exact match)
-  representing [Jaro](https://en.wikipedia.org/wiki/Jaro–Winkler_distance)
+  Computes the bag distance of two strings.
+
+  Returns a float value between 0 and 1 representing the bag
+  distance between `string1` and `string2`.
+
+  The bag distance is meant to be an efficient approximation
+  of the distance between two strings to quickly rule out strings
+  that are largely different.
+
+  The algorithm is outlined in the "String Matching with Metric
+  Trees Using an Approximate Distance" paper by Ilaria Bartolini,
+  Paolo Ciaccia, and Marco Patella.
+
+  ## Examples
+
+      iex> String.bag_distance("abc", "")
+      0.0
+      iex> String.bag_distance("abcd", "a")
+      0.25
+      iex> String.bag_distance("abcd", "ab")
+      0.5
+      iex> String.bag_distance("abcd", "abc")
+      0.75
+      iex> String.bag_distance("abcd", "abcd")
+      1.0
+
```

Comment:
What do you think about including a note mentioning this function uses graphemes to calculate the distance?

```elixir
iex> String.bag_distance("éṩab", "éṩa")
0.75
```

Reply:
Nevermind. I see almost all the functions in this module use graphemes (as mentioned in the documentation of the module), and their documentation do not include any references to graphemes.